File: src/util/file/FileSort.java
package util.file;

import util.Sort;

/**
 * Denotes {@link java.io.File} sort: directory first, file first or no preference.
 */
public enum FileSort {
    DIR_FIRST(Sort.DESCENDING),
    FILE_FIRST(Sort.ASCENDING),
    NONE(Sort.NONE);

    public final Sort sort;

    FileSort(Sort s) {
        sort = s;
    }
}

File: src/util/main/sp/it/util/file/FileSort.kt
package sp.it.util.file

import sp.it.util.Sort

/** Denotes [java.io.File] by type sort: directory first, file first or no preference. */
enum class FileSort constructor(val sort: Sort) {
   DIR_FIRST(Sort.DESCENDING),
   FILE_FIRST(Sort.ASCENDING),
   NONE(Sort.NONE)
}